source/module_pw/pw_basis.h

```diff
@@ -260,6 +260,13 @@ class PW_Basis
     using resmem_int_op = psi::memory::resize_memory_op<int, psi::DEVICE_GPU>;
     using delmem_int_op = psi::memory::delete_memory_op<int, psi::DEVICE_GPU>;
     using syncmem_int_h2d_op = psi::memory::synchronize_memory_op<int, psi::DEVICE_GPU, psi::DEVICE_CPU>;
+
+    void set_device(std::string device_);
+    void set_precision(std::string precision_);
+
+protected:
+    std::string device = "cpu";
+    std::string precision = "double";
```

Comment:
![1](https://user-images.githubusercontent.com/76200646/221788567-1238976a-65de-4844-8838-d96011e88333.png)
Please remove globalV::precision_flag = "single" in pw_test.cpp. 
Change `std::string precision = "double"` to
`
#ifdef __ENABLE_FLOAT_FFTW

    std::string precision = "single"

#else

    std::string precision = "double"

#endif
`
Then the segment fault can be solved.

Reply:
Fixed